File: Translators/src/main/java/artofillusion/translators/PLYTranslator.java
package artofillusion.translators;

import artofillusion.Scene;
import artofillusion.Translator;
import buoy.widget.BFrame;

public class PLYTranslator implements Translator {
    /**
     * Get the name of the file format which this translator imports or exports.
     */
    @Override
    public String getName() {
        return "Stanford PLY";
    }

    /**
     * Specify whether this translator can import files.
     */
    @Override
    public boolean canImport() {
        return true;
    }

    /**
     * Prompt the user to select a file, read it, and create a new LayoutWindow containing
     * the imported scene. parent is the Frame which should be used as the parent for
     * dialog boxes. If canImport() returns false, this method will never be called.
     *
     * @param parent
     */
    @Override
    public void importFile(BFrame parent) {
        PLYImporter.importFile(parent);
    }

    /**
     * Prompt the user for a filename and any other necessary information, and export the
     * scene. parent is the Frame which should be used as the parent for dialog boxes.
     * The user should be given the option of only exporting the objects which are
     * currently selected. If canExport() returns false, this method will never be called.
     *
     * @param parent
     * @param theScene
     */
    @Override
    public void exportFile(BFrame parent, Scene theScene) {
        //Operation not yet supported
    }

    /**
     * Specify whether this translator can export files.
     */
    @Override
    public boolean canExport() {
        return false;
    }
}

File: Translators/src/main/kotlin/artofillusion/translators/PLYTranslator.kt
/* Copyright (C) 2025 by Maksim Khramov

   This program is free software; you can redistribute it and/or modify it under the
   terms of the GNU General Public License as published by the Free Software
   Foundation; either version 2 of the License, or (at your option) any later version.

   This program is distributed in the hope that it will be useful, but WITHOUT ANY
   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */

package artofillusion.translators

import artofillusion.Scene
import artofillusion.Translator
import artofillusion.ui.Translate
import buoy.widget.BFrame
import javax.swing.JFileChooser
import javax.swing.filechooser.FileNameExtensionFilter

class PLYImporter {
    companion object {
        fun importFile(parent: BFrame?) {
            val chooser = JFileChooser();
            val objFilter = FileNameExtensionFilter(Translate.text("fileFilter.ply"), "ply");
            chooser.addChoosableFileFilter(objFilter)
            chooser.isAcceptAllFileFilterUsed = true
            chooser.fileFilter = objFilter
            chooser.showOpenDialog(parent?.component);

        }
    }
}

class PLYTranslator: Translator {
    override fun getName(): String? = "Stanford Triangle Format (.ply)"

    override fun importFile(parent: BFrame?) = PLYImporter.importFile(parent);

    override fun canExport(): Boolean = false

    override fun exportFile(parent: BFrame?, theScene: Scene?) {
    }
}